src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs

```diff
@@ -459,6 +534,168 @@ private static CompressionOption GetCompressionOptionFromZipFileInfo()
+                {
+                    pieceInfo.ZipArchiveEntry.Delete();
+                }
+            }
+            //Its okay for us to not clean up the sortedPieceInfoList datastructure, as the
+            //owning part is about to be deleted.
+        }
+
+        /// <summary>
+        /// An auxiliary function of GetPartsCore, this function sorts out the piece name
+        /// descriptors accumulated in pieceNumber into valid piece sequences and garbage
+        /// (i.e. ignorable Zip items).
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The procedure used relies on 'pieces' members to be sorted lexicographically
+        /// on &lt;name, number, isLast> triples, with name comparisons being case insensitive.
+        /// This is enforced by PieceInfo's IComparable implementation.
+        /// </para>
+        /// </remarks>
+        private void ProcessPieces(SortedSet<ZipPackagePartPiece> pieceSet, List<PackagePart> parts)
+        {
+            // The zip items related to the ContentTypes.xml should have been already processed.
+            // Only those zip items that follow the valid piece naming syntax and have a valid
+            // part name should show up in this list.
+            // piece.PartUri should be non-null
+
+            // Exit if nothing to do.
+            if (pieceSet.Count == 0)
+                return;
+
+            string? normalizedPrefixNameForCurrentSequence = null;
+            int startIndexOfCurrentSequence = 0; // Value is ignored as long as
+                                                 // prefixNameForCurrentSequence is null.
+
+            List<ZipPackagePartPiece> pieces = new List<ZipPackagePartPiece>(pieceSet);
+
+            for (int i = 0; i < pieces.Count; ++i)
+            {
+                // Looking for the start of a sequence.
+                if (normalizedPrefixNameForCurrentSequence == null)
+                {
+                    if (pieces[i].PieceNumber != 0)
+                    {
+                        // Whether or not this piece bears the same unsuffixed name as a complete
+                        // sequence just processed, it has to be ignored without reporting an error.
+                        _ignoredItemHelper.AddItemForStrayPiece(pieces[i]);
+                        continue;
+                    }
+                    else
+                    {
+                        // Found the start of a sequence.
+                        startIndexOfCurrentSequence = i;
+                        normalizedPrefixNameForCurrentSequence = pieces[i].NormalizedPrefixName;
+                    }
+                }
+
+                // Not a start piece. Carry out validity checks.
+                else
+                {
+                    //Check for incomplete sequence.
+                    if (string.CompareOrdinal(pieces[i].NormalizedPrefixName, normalizedPrefixNameForCurrentSequence) != 0)
```

Comment:
Use `!=` for ordinal equality comparison

Reply:
Yes, agreed.